File: kolokwia/K1_162681/court.py
import datetime


class Court:
    def __init__(self, width: float = 68, length: float = 150, address: str = "", year_built: int = 0) -> None:
        if 45 > width or width > 90 or 90 > length or length > 120:
            width = 70
            length = 75
            self.__width = width
            self.__length = length
        else:
            self.__width = width
            self.__length = length
        self.__address = address
        self.__year_built = year_built

    @property
    def width(self) -> float:
        return self.__width

    @width.setter
    def width(self, value: float) -> None:
        if 45 > value or value > 90:
            self.__width
        else:
            self.__width = value

    @property
    def length(self) -> float:
        return self.__length

    @length.setter
    def length(self, value: float) -> None:
        if 90 > value or value > 120:
            self.__length
        else:
            self.__length = value

    @property
    def address(self) -> str:
        return self.__address

    @address.setter
    def address(self, value: str) -> None:
        self.__address = value

    @property
    def year_built(self) -> int:
        return self.__year_built

    @year_built.setter
    def year_built(self, value: int) -> None:
        self.__year_built = value

    def area(self) -> float:
        return self.__width * self.__length

    def __str__(self):
        return f'Boisko wybudowane w roku {self.year_built}, o długości {self.length} metrów i ' \
               f'szerokości {self.width} metrów. \n' \
               f'Pole powierzchni: {self.area()} mkw. \n' \
               f'Adres: {self.address}.'

    def __eq__(self, other):
        return self.area() == other.area()

    def __ne__(self, other):
        return self.area() != other.area()


def validate(court: Court) -> None:
    year = datetime.datetime.today().year
    if 0 < court.year_built > year:
        court.year_built = year

File: Java/Kolos/Kolos/src/jacek/sosnowski/paczka/Court.java
package jacek.sosnowski.paczka;

import java.time.LocalDate;
import java.util.Objects;

class Court {
    private double width;
    private double length;
    private String address;
    private int year_built;

    public Court(double width, double length, String address, int year_built){
        if (length < 90 ||  length > 120)
        {
            length = 150;
        }
        if (width < 45 || width > 90)
        {
            width = 68;
        }
        this.width = width;
        this.length = length;
        this.address = address;
        this.year_built = year_built;
    }
    public double getWidth() {
        return this.width;
    }

    public double getLength() {
        return this.length;
    }

    public String getAddress() {
        return this.address;
    }

    public int getYear_built() {
        return this.year_built;
    }

    public void setWidth(double width) {
        if (width >= 45 && width <= 90)
            this.width = width;
    }

    public void setLength(double length) {
        if (length >= 90 && length <= 120)
            this.length = length;
    }

    public void setAddress(String address) {
        this.address = address;
    }

    public void setYear_built(int year_built) {
        this.year_built = year_built;
    }

    public double area(){
        return this.width * this.length;
    }

    public static void validate(Court other){
        if (other.year_built < 0 || other.year_built > LocalDate.now().getYear()){
            other.year_built = LocalDate.now().getYear();
        }
    }

    @Override
    public String toString(){
        return "Boisko wybudowane w roku " + this.year_built + ", o długości " + this.length + " metrów i szerokości " + this.width + " metrów.\n" +
                "Pole powierzchni: " + this.area() + " mkw.\n" +
                "Adres: " + this.address + ".";
    }

    @Override
    public boolean equals(Object obj) {
        if (this == obj)
            return true;
        if (!(obj instanceof Court))
            return false;
        Court court = (Court) obj;
        return this.area() == court.area();
    }

    @Override
    public int hashCode() {
        return Objects.hash(width, length, address, year_built);
    }
}
